src/ServiceManagement/Compute/Commands.ServiceManagement/Extensions/Common/ExtensionRole.cs

```diff
@@ -63,12 +78,16 @@ public override string ToString()
 
         public string GetExtensionId(string extensionName, string slot, int index)
         {
-            var normalizedExtName = extensionName.Replace(".", string.Empty);
-            normalizedExtName = normalizedExtName.Replace(" ", string.Empty);
-            normalizedExtName = normalizedExtName.Replace("_", string.Empty);
+            var normalizedExtName = RemoveDisallowedCharacters(extensionName);
 
             var suffix = new StringBuilder();
             suffix.AppendFormat(ExtensionIdSuffixTemplate, normalizedExtName, slot, index);
+            if (suffix.Length > MaxSuffixLength)
+            {
+                int lenDiff = suffix.Length - MaxSuffixLength;
+                int startIndex = 1; // Suffix starts with '-'
+                suffix.Remove(startIndex + normalizedExtName.Length - lenDiff, lenDiff);
```

Comment:
what do you want to achieve here? string.Remove returns a new string which you do not use anywhere.